aronnax.f90

```diff
@@ -1694,21 +1694,21 @@ subroutine Ext_solver(MPI_COMM_WORLD, hypre_A, hypre_grid, num_procs, &
     end do
   end do
 
```

Comment:
This commit is a teachable moment.

1. "by all accounts didn't work as intended anyway" is not a very helpful phrase to see in a log message.  What accounts?  What was the reported problem?  What evidence blames this particular part of the code?

2. Turning a loop that should execute exactly once (there being one process) into just invoking the loop body one time shouldn't be a solution to anything, because it shouldn't have any effect.  If you suspected that the loop was iterating more than once, a simple `print *, num_procs` or `print *, i` should have assuaged your concern.

2a. Aside: Why is this thing looping over the number of processes anyway?  Shouldn't it just be filling its own box?  But perhaps I don't understand MPI/Hypre yet.

3. The actual change here is bogus: instead of looping over all i from 0 to num_procs-1 inclusive, the code as written _uses the old value of i from the enclosing scope_, which by dumb luck happens to be defined.  That the compiler accepted this is a consequence of this barbarous programming language forcing one to declare one's loop variables in advance, instead of defining them on the spot in the looping construct and taking them back out of scope when the looping construct ends, like all civilized systems do (for future reference, Python is not civilized in this sense either).

4. In this instance, `i` happens to be `nx+1 = 11`, which has the effect of reading the box boundaries from somewhere off the ends of the `ilower` and `iupper` arrays, i.e., making them up from whole cloth.  When I tried it, I got `ilower(11,1) ~ 32700, ilower(11,2) = 32, iupper(11,1) ~ 1.95 billion, iupper(11,2) ~ 1.91 billion`. (Interestingly, they were the same in every time step, but varied somewhat across runs.  I suppose it's actually reading from some nearby array that may have some floating-point input data or something.)  I guess this causes Hypre to decide that you are setting values for an empty box, and leave `hypre_b` and `hypre_x` at some default value, like 0.  Conversely, at extraction time, I guess Hypre simply doesn't modify the `values` array at all, and, though I haven't checked this, I guess this whole routine is now equivalent to `etanew = etastar/dt**2`.  This, in turn, stops Hypre from complaining about NaNs in its inputs, and manifests as the test suite reporting "wrong answer".  Commenting out the three calls to VectorSet/GetBoxValues entirely also exhibits the same external behavior.

5. Undoing this change and adding some more debug prints instead suggests that what's actually happening to cause those Hypre complaints is that the simulation blows up numerically over multiple iterations, eventually leading to NaNs, which Hypre catches before `break_if_NaN` does because the latter is only called at dump time.  That's an argument in favor of `break_if_NaN` at every timestep.

Possible ways to proceed from here:
- Revert this commit regardless.
- Could try to further debug the simulation by "force of will" (i.e., adding print statements and trying to figure out what's happening)
- Could step back and try to develop more effective test inspection.  A one-number summary of the discrepancy in the first time point that has a discrepancy is not very informative; perhaps the test suite should make some visualizations of the differences when it fails.
- Will the `pysics-tests` branch provide such debugging tools?  Should we merge Hypre as still-broken-but-probably-fixable "experimental code" so that we can develop the physics-based testing well enough to figure out what's happening here?